script/lint_git

```diff
@@ -19,4 +19,13 @@ else
   cache store $cache_key $HOME/bin/lintje
 fi
 
-$HOME/bin/lintje $SEMAPHORE_GIT_COMMIT_RANGE
+# When force-pushing, the commit range may begin at a commit that does not
+# exist. When this happens, check only the commit at the head of the range.
```

Comment:
The end commit will always be found, because it is the head of what was just force-pushed into existence. What can be missing is the start commit, because it can be the previous head of what was just force-pushed into oblivion.

> I don't know if we want to fall back of anything if it doesn't match the same range.

I agree. I'm happy to echo some message like "Lintje can't run because the commit range given points to commits that don't exist" instead and `exit 0`. I thought checking the head commit of the given range (which will always exist) was better than nothing, but it's probably best to be explicit about what's happening.

> So far I haven't seen Lintje fail like this unless the commits aren't present anymore on the branch (because of a force push).

Yes, that is when it fails in this manner. If Lintje is given a range that starts at a non-existing commit, then it can't run `git log` on it, so it can't run.

> Have you noticed it also runs into this error on the last forced pushed version of the branch?

I think it depends on what the last force push changed from the previous state of the branch. I've only seen it happen when commits are replaced entirely -- I attempted to explain when I think it happens on the commit description. It may not be Semaphore's fault at all, as it might be getting these values from Github.

Reply:
I'll come back to this PR when my head is a better place to review this.

Reply:
Likewise. I'll change it to a draft for now, we can revisit it later.